app/models/manageiq/providers/ansible_tower/configuration_manager/refresh_worker/parser.rb

```diff
@@ -0,0 +1,77 @@
+class ManageIQ::Providers::AnsibleTower::ConfigurationManager::RefreshWorker::Parser
+  attr_accessor :inventory, :ems
+
+  def initialize(ems, target, inventory)
+    @ems    = ems
+    @target = target
+    @inventory = inventory
+  end
+
+  def parse
+    [
+      inventory_groups,
+      configured_systems,
+      configuration_scripts,
+    ]
+  end
+
+  def inventory_groups
+    # FIXME: memoization  should not be needed.
```

Comment:
well, I've started to store them in Inventory under AWS, but probably that is not ideal